cmd/pj-rehearse/main.go

```diff
@@ -143,29 +135,46 @@ func main() {
 		}
 	}
 
-	prowConfig, prowPRConfig, err := getProwConfigs(o.candidatePath, jobSpec.Refs.BaseSHA)
+	prConfig := config.GetAllConfigs(o.releaseRepoPath, logger)
+	masterConfig, err := config.GetAllConfigsFromSHA(o.releaseRepoPath, jobSpec.Refs.BaseSHA, logger)
 	if err != nil {
-		logger.WithError(err).Error("could not load prow configs")
+		logger.WithError(err).Error("could not load configuration from base revision of release repo")
 		gracefulExit(o.noFail, misconfigurationOutput)
 	}
 
-	ciopConfig, ciopPRConfig, err := getCiopConfigs(o.candidatePath, jobSpec.Refs.BaseSHA)
-	if err != nil {
-		logger.WithError(err).Error("could not load ci-operator configs")
+	// We always need both Prow config versions, otherwise we cannot compare them
+	if masterConfig.Prow == nil || prConfig.Prow == nil {
+		logger.WithError(err).Error("could not load Prow configs from base or tested revision of release repo")
+		gracefulExit(o.noFail, misconfigurationOutput)
+	}
+	// We always need PR versions of templates and ciop config, otherwise we cannot provide them to rehearsed jobs
+	if prConfig.Templates == nil || prConfig.CiOperator == nil {
+		logger.WithError(err).Error("could not load template/ci-operator configs from tested revision of release repo")
 		gracefulExit(o.noFail, misconfigurationOutput)
 	}
 
-	pjclient, err := rehearse.NewProwJobClient(clusterConfig, prowConfig.ProwJobNamespace, o.dryRun)
+	// We can only detect changes if we managed to load both ci-operator config versions
+	if masterConfig.CiOperator != nil && prConfig.CiOperator != nil {
+		changedCiopConfigs := diffs.GetChangedCiopConfigs(masterConfig.CiOperator, prConfig.CiOperator, logger)
+		for name := range changedCiopConfigs {
+			logger.WithField("ciop-config", name).Info("Changed ci-operator config")
+		}
+	}
+
+	// We can only detect changes if we managed to load both CI template versions
+	if masterConfig.Templates != nil && prConfig.Templates != nil {
```

Comment:
this can be checked also in `GetAllConfigs`.

Reply:
This is exactly the place that cannot - `GetAllConfigs` only sees a single revision of the config, and it does not know anything about comparison of templates.